packages/babel-plugin-transform-typescript/src/index.js

```diff
@@ -74,7 +74,11 @@ export default declare(
             );
           }
 
-          path.remove();
+          if (node.definite && node.decorators) {
+            node.definite = null;
+          } else {
+            path.remove();
+          }
```

Comment:
nit: I feel like it's clearer to only check against `!node.decorators` here (like we did in the `else if` branch).
```js
if (!node.decorators) {
  path.remove();
}
```
and then reset `definite` later (like we did for `abstract`, `accessibility`, etc.)
```js
if (node.definite) node.definite = null;
```

Reply:
that's a *definite* improvement in clarity 😉